task1_commit1

BespoyasovaV opened the pull request.

BespoyasovaV committed:
task1_commit1

BespoyasovaV committed:
task1_commit1

lyaska commented on src/main/java/ru/documents/Document.java:
```diff
@@ -0,0 +1,107 @@
+package ru.documents;
+
+import java.util.Comparator;
+
```
нет javadoc-ов

- [ ]

lyaska replied:
и автора класса укажи

lyaska commented on src/main/java/ru/documents/Document.java:
```diff
@@ -0,0 +1,107 @@
+
+import java.util.Comparator;
+
+public class Document  {
```
не запушила файлы с классами других документов

lyaska replied:
разложи все по пакетам, сущности к сущностям, фабрику к фабрике и тд

lyaska commented on src/main/java/ru/documents/Document.java:
```diff
@@ -0,0 +1,107 @@
+
+
+
+
```
Почему так много пробелов и новых строк ctrl+alt+l

lyaska commented on src/main/java/ru/documents/DocumentFactory.java:
```diff
@@ -0,0 +1,49 @@
+import ru.documents.Task;
+import ru.documents.DocumentCreate;
+
+public class DocumentFactory extends RandomValue {
```
javadoc

lyaska replied:
что за класс randomValue?

lyaska commented on src/main/java/ru/documents/DocumentExistException.java:
```diff
@@ -0,0 +1,13 @@
+package ru.documents;
+
+public class DocumentExistException extends Exception{
```
javadoc

lyaska commented on src/main/java/ru/documents/DocumentFactory.java:
```diff
@@ -0,0 +1,49 @@
+public class DocumentFactory extends RandomValue {
+
+
+
```
пробелы, пропуски.... сделай код чище и удобочитаемым

lyaska commented on src/main/java/MainClass.java:
```diff
@@ -0,0 +1,29 @@
+            doc[i].conclusion();
+            doc[i].norm_concl();
+
+
```
пробелы

lyaska commented on src/main/java/ru/documents/DocumentFactory.java:
```diff
@@ -0,0 +1,49 @@
+        if (documentType == null) {
+            return null;
+        }
+        if (documentType.equalsIgnoreCase("Task")) {
```
Определять тип документа это не задача этого класса

lyaska commented on src/main/java/ru/documents/DocumentFactory.java:
```diff
@@ -0,0 +1,49 @@
+            return null;
+        }
+        if (documentType.equalsIgnoreCase("Task")) {
+            Task t = new Task(generateid(), generate_name_doc(), generateName_text(), getReg_num_for_Task(), generate_date(), generatePerson(), generate_date_issue(), generate_term(), generate_fio_executor(), generate_sign(), generate_controller());
```
Почитай еще раз про паттерн абстрактная фабрика, у тебя должен быть абстрактный класс, который подходит для создания всех документов. Для этого можешь использовать дженерики

lyaska replied:
у тебя будет абстрактный класс по созданию документов, интерфейс с методом который создает документ и три класса фабрик для каждого типа документа, который наследуется от фабрики и имплементирует интерфейс

lyaska commented on src/main/java/ru/documents/DocumentFactory.java:
```diff
@@ -0,0 +1,49 @@
+
+        }
+        if (documentType.equalsIgnoreCase("Incoming")) {
+            Incoming i = new Incoming(generateid(), generate_name_doc(), generateName_text(), getReg_num_for_Incoming(), generate_date(), generatePerson(), generate_fio_sender(), generate_fio_executor(), generate_ish_num(), generate_date_issue());
```
названия переменных лучше писать более читаемыми и развернутыми

lyaska commented on src/main/java/ru/documents/DocumentFactory.java:
```diff
@@ -0,0 +1,49 @@
+
+    public void parse() {
+
+
```
зачем этот метод?

lyaska commented on src/main/java/ru/documents/Document.java:
```diff
@@ -0,0 +1,107 @@
+
+
+    class DocumentDateRegComparator implements Comparator<Document> {
+        public int compare(Document a, Document b) {
```
зачем лишняя вложенность классов? Comparator реализует только класс Document  и в одном методе сравнивает документы по номеру и дате. почитай про ключевое слово this

lyaska commented on .idea/uiDesigner.xml:
```diff
@@ -0,0 +1,124 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
этот файл не нужен в пр-е, убери

albertino90 replied:
еще убери всю папку target из ПРа. Добавь в gitignore

lyaska commented on src/main/java/MainClass.java:
```diff
@@ -0,0 +1,29 @@
+import ru.documents.DocumentFactory;
+
+public class MainClass {
+    public static void main(String[] args) throws DocumentExistException {
```
это исключение должно обрабатываться при создании документа, не нужно его выбрасывать здесь

lyaska commented on src/main/java/MainClass.java:
```diff
@@ -0,0 +1,29 @@
+
+public class MainClass {
+    public static void main(String[] args) throws DocumentExistException {
+
```
javadoc

lyaska commented on src/main/java/ru/documents/DocumentExistException.java:
```diff
@@ -0,0 +1,13 @@
+    private String number;
+
+    public String getNumber() {
+        System.out.println("вызвано исключение, совпадают номера id");
```
не нужно

lyaska commented on src/main/java/ru/documents/DocumentExistException.java:
```diff
@@ -0,0 +1,13 @@
+        return number;
+    }
+    public DocumentExistException(String number ){
+        System.out.println("вызвано исключение, совпадают номера id");
```
класс является наследником от Exception воспользуйся конструктором родителя для вывода сообщения

lyaska commented on src/main/java/ru/documents/DocumentExistException.java:
```diff
@@ -0,0 +1,13 @@
+package ru.documents;
+
+public class DocumentExistException extends Exception{
+    private String number;
```
не нужно

lyaska commented on src/main/java/ru/documents/DocumentFactory.java:
```diff
@@ -0,0 +1,49 @@
+
+
+    public DocumentFactory() {
+        System.out.println("Создана фабрика");
```
это нужно?

lyaska commented on src/main/java/ru/documents/DocumentFactory.java:
```diff
@@ -0,0 +1,49 @@
+        System.out.println("Создана фабрика");
+    }
+
+    public DocumentCreate getDocument(String documentType) throws DocumentExistException {
```
исключение нужно не выбрасывать а обрабатывать при создании документа

lyaska commented on src/main/java/MainClass.java:
```diff
@@ -0,0 +1,29 @@
+public class MainClass {
+    public static void main(String[] args) throws DocumentExistException {
+
+        String[] arr = new String[]{"Task", "Incoming", "Outgoing"};
```
лучше вынести

lyaska commented on src/main/java/MainClass.java:
```diff
@@ -0,0 +1,29 @@
+        DocumentCreate[] doc = new DocumentCreate[rand_number];
+        System.out.println(rand_number);
+        for (int i = 0; i < rand_number; i++) {
+            int rand_number_of_doc = (int) (Math.random() * 3);
```
заниматься рандомом это не обязанность этого класса, вынеси

lyaska commented on src/main/java/ru/documents/DocumentCreate.java:
```diff
@@ -0,0 +1,11 @@
+public interface DocumentCreate {
+    void create();
+
+    void conclusion();
```
В этом интерфейсе только методы для создания документов

lyaska commented on src/main/java/ru/documents/DocumentCreate.java:
```diff
@@ -0,0 +1,11 @@
+package ru.documents;
+
+public interface DocumentCreate {
```
javadoc

lyaska commented on src/main/java/ru/documents/Incoming.java:
```diff
@@ -0,0 +1,96 @@
+
+import ru.documents.Document;
+
+public class Incoming extends Document implements DocumentCreate {
```
Этот интерфейс здесь не нужен

lyaska commented on src/main/java/ru/documents/Incoming.java:
```diff
@@ -0,0 +1,96 @@
+    int outgoing_num;
+    String outgoing_date;
+    String name_first="Входящий";
+    static int check_num=0;
```
Зачем статик? Пусть будет обычное поле

lyaska commented on src/main/java/ru/documents/Incoming.java:
```diff
@@ -0,0 +1,96 @@
+    //constructor
+    public Incoming(int id, String name, String text, int reg_num, String date_reg, Person author, String sender, String destination, int outgoing_num, String outgoing_date)
+    {
+
```
пробелы

lyaska commented on src/main/java/ru/documents/Incoming.java:
```diff
@@ -0,0 +1,96 @@
+    @Override
+    public String toString() {
+        return "Отправитель: "+sender+
+                "Адресат: "+ destination+
```
используй MessageFormat

lyaska commented on src/main/java/ru/documents/Incoming.java:
```diff
@@ -0,0 +1,96 @@
+    public void conclusion() {
+
+        System.out.println("      " + author.name+" "+ author.Surname+" "+author.patronymic);
+        //message format
```
Зачем? ты же переопределила toString

lyaska commented on src/main/java/ru/documents/Incoming.java:
```diff
@@ -0,0 +1,96 @@
+    @Override
+    public void norm_concl() {
+
+        System.out.print(name_first + " " + "№" + getCheck_num() + " " + "от" + " " + getDate_reg() + " " + getName());
```
то же самое

lyaska commented on src/main/java/ru/documents/Person.java:
```diff
@@ -0,0 +1,56 @@
+    }
+
+    public Person( String surname,String name,  String patronymic) {
+
```
ctrl+alt+l

lyaska commented on src/main/java/ru/documents/RandomValue.java:
```diff
@@ -0,0 +1,184 @@
+
+import java.util.ArrayList;
+
+public class RandomValue {
```
javadoc

lyaska commented on src/main/java/ru/documents/RandomValue.java:
```diff
@@ -0,0 +1,184 @@
+
+    String[] fioExecutorList = new String[]{"Карамов Валерий Степанов", "Полянская Карина Аликовна", "Березнянский Никита Вячеславович", "Подорожная Светлана Петровна"};
+    String[] fioSenderList = new String[]{"Бесполова Катерина Никитична", "Никитина Юлия Игоревна", "Галлямутдинова Камилла Ринатовна", "Латипов Александр Сергеевич"};
+    String[] DeliveryList = new String[]{"Самовывоз", "Курьером", "ПочтаРоссии"};
```
это вроде должен быть enum со значениями

lyaska commented on src/main/java/ru/documents/RandomValue.java:
```diff
@@ -0,0 +1,184 @@
+    String[] fioList = new String[]{"Барсуков Сергей Геннадиевич", "Полянская Кристина Викторовна", "Березнянкова Анфиса Кирилловна", "Киреева Надежда Семенова", "Бесполова Катерина Никитична", "Никитина Юлия Игоревна", "Галлямутдинова Камилла Ринатовна", "Латипов Александр Сергеевич"};
+
+    String[] fioExecutorList = new String[]{"Карамов Валерий Степанов", "Полянская Карина Аликовна", "Березнянский Никита Вячеславович", "Подорожная Светлана Петровна"};
+    String[] fioSenderList = new String[]{"Бесполова Катерина Никитична", "Никитина Юлия Игоревна", "Галлямутдинова Камилла Ринатовна", "Латипов Александр Сергеевич"};
```
может лучше взять одну коллекцию с юзерами?

lyaska commented on src/main/java/ru/documents/RandomValue.java:
```diff
@@ -0,0 +1,184 @@
+    public ArrayList<Person> person_list() {
+
+        person_list.add(new Person("Барсуков", "Сергей", "Геннадиевич"));
+        person_list.add(new Person("Березнянкова", " Анфиса", "Кирилловна"));
```
через static ты можешь заполнять коллекцию при создании класса

lyaska commented on src/main/java/ru/documents/RandomValue.java:
```diff
@@ -0,0 +1,184 @@
+
+
+    static int reg_num_for_Task = 0;
+    static int reg_num_for_Incomming = 0;
```
зачем? можно просто давать рандомный номер всем документам, необязательно что бы они шли по порядку

lyaska commented on src/main/java/ru/documents/RandomValue.java:
```diff
@@ -0,0 +1,184 @@
+    public int generateid() throws DocumentExistException {
+        int id = (int) (Math.random() * 100);
+        if (check_id.contains(id)) {
+            throw new DocumentExistException("вызвано исключение, совпадают номера id");
```
это исключение вроде для номеров документа

lyaska commented on src/main/java/ru/documents/RandomValue.java:
```diff
@@ -0,0 +1,184 @@
+    }
+
+    public String generate_date() {
+        java.util.Random day = new java.util.Random();
```
зачем два метода которые по сути делают одно и тоже - создают дату?

lyaska commented on src/main/java/ru/documents/RandomValue.java:
```diff
@@ -0,0 +1,184 @@
+
+    public int generate_term() {
+        int term = (int) (Math.random() * 56);
+
```
пробелы

lyaska commented on src/main/java/ru/documents/RandomValue.java:
```diff
@@ -0,0 +1,184 @@
+
+    public Person generatePerson() {
+
+        int n = (int) (Math.random() * 5);
```
создай один метод, который сможет доставать рандомный элемент из любой коллекции и используй его везде

BespoyasovaV committed:
25.01.2021

BespoyasovaV committed:
25.01.2021

BespoyasovaV committed:
javadoc.
Исправила некоторые замечания.
Наконец-то получилось помещать авторов и их документы в map.
Исправила исключение, теперь оно вызывается на повторяющийся регистрационный номер, а не на Id. Убрала ненужные методы и коллекции.

lyaska commented on src/main/java/MainClass.java:
```diff
@@ -1,30 +1,29 @@
-            doc[i].toString();
-
-
+            //doc[i].toString();
```
удалить закомментированное

lyaska commented on src/main/java/MainClass.java:
```diff
@@ -1,30 +1,29 @@
         }
 
         System.out.println("***************************");
+        System.out.println();
```
зачем все эти звездочки и пробелы?

lyaska commented on src/main/java/ru/documents/Task.java:
```diff
@@ -2,37 +2,42 @@
+    ArrayList<Integer> check_id = new ArrayList<Integer>();
+
+    //getters
+    private int getCheck_num() throws DocumentExistException {
```
генерацией номеров занимается фабрика а не сам класс документа. В классе документа должны быть только поля класса, геттеры/сеттеры и метод toString

lyaska commented on src/main/java/ru/documents/factory/DocumentCreate.java:
```diff
@@ -1,9 +1,10 @@
 
 public interface DocumentCreate {
-    void create();
     String toString();
```
зачем это интерфейсу?

lyaska commented on src/main/java/ru/documents/factory/DocumentFactory.java:
```diff
@@ -1,45 +1,67 @@
+/**
+ * класс создает документы Task, Incoming,Outgoing и, при создании, находит ему место в мапе под нужными ФИО
+ */
```
джавадок пишется над названием класса, а не над импортами. И еще пиши автора @author

lyaska commented on src/main/java/ru/documents/factory/DocumentFactory.java:
```diff
@@ -1,45 +1,67 @@
+    public HashMap<String, ArrayList<Document>> person_and_document = new HashMap<String, ArrayList<Document>>();
 
-    public DocumentCreate getDocument(String documentType) throws DocumentExistException {
+    public DocumentCreate getDocument(String documentType) {
```
Определять тип документа лучше в отдельном классе, и нужно для каждого типа документа свой класс фабрику создать, который будет создавать документы только одного типа.

lyaska commented on src/main/java/ru/documents/factory/DocumentFactory.java:
```diff
@@ -1,45 +1,67 @@
         }
+
         return null;
 
```
пробелы

lyaska commented on src/main/java/ru/generate/RandomValue.java:
```diff
@@ -1,21 +1,16 @@
-import java.util.HashMap;
+import java.util.Random;
 
 public class RandomValue {
```
джавадоки везде напиши

lyaska commented on src/main/java/ru/generate/RandomValue.java:
```diff
@@ -24,55 +19,23 @@
 
-
-    enum Delivery {
+    public enum Delivery {
```
enum вынеси в отдельный файл, погугли как правильно enum-ы оформлять

lyaska commented on src/main/java/ru/generate/RandomValue.java:
```diff
@@ -24,55 +19,23 @@
-        return person_list;
-    }
-
     public int generate_rand_number() {
```
вообще по кодстайлу нужно название переменных и методов писать типа generateRandomValue, те используется lowerCamelCase

BespoyasovaV committed:
Новая абстрактная фабрика, javadoc, исправленные назнания переменных и методов,

lyaska commented on src/main/java/ru/documents/factory/AbstractFactory.java:
```diff
@@ -0,0 +1,22 @@
+
+/**
+ * заполянет поля документа и возвращает его
+ */
```
Добавь автора к javadoc для классов @author vbespoyasova. Ну и пиши с заглавной буквы, красивей так

lyaska commented on src/main/java/ru/documents/factory/AbstractFactory.java:
```diff
@@ -0,0 +1,22 @@
+ */
+public abstract class AbstractFactory<T extends Document> extends RandomValue {
+    void getDocument(Document document) {
+
```
удали лишние пробелы, пропуски строк, наведи в пр-е красоту

lyaska commented on src/main/java/ru/documents/factory/Conclusion.java:
```diff
@@ -0,0 +1,7 @@
+import ru.documents.Document;
+
+public interface Conclusion <T extends Document> {
+   String toString(T document);
```
этот интерфейс нужен?

lyaska commented on src/main/java/ru/documents/factory/AbstractFactory.java:
```diff
@@ -0,0 +1,22 @@
+ * заполянет поля документа и возвращает его
+ */
+public abstract class AbstractFactory<T extends Document> extends RandomValue {
+    void getDocument(Document document) {
```
Над методами тоже пишутся javadoc по такому же принципу, только без указания автора.

lyaska commented on src/main/java/ru/documents/factory/DocumentCreate.java:
```diff
@@ -0,0 +1,13 @@
+ * от этого класса наследуются фабрики, которые оздают документы разных видов
+ */
+public interface DocumentCreate <T extends  Document> {
+    T create();
```
javadoc

lyaska commented on src/main/java/ru/documents/factory/IncomingFactory.java:
```diff
@@ -0,0 +1,32 @@
+/**
+ * класс создает документ Incoming, но без полей родителя
+ */
+public class IncomingFactory extends AbstractFactory<Incoming> implements Conclusion,DocumentCreate<Incoming> {
```
интерфейс Conclusion не нужен

lyaska commented on src/main/java/ru/documents/factory/IncomingFactory.java:
```diff
@@ -0,0 +1,32 @@
+        return new Incoming();
+    }
+
+    @Override
```
{@inhreitdoc}

lyaska commented on src/main/java/ru/documents/factory/IncomingFactory.java:
```diff
@@ -0,0 +1,32 @@
+
+    @Override
+    public String toString(Document document) {
+        document=(Incoming) document;
```
Этот метод здесь не нужен

lyaska commented on src/main/java/ru/documents/factory/OutgoingFactory.java:
```diff
@@ -0,0 +1,30 @@
+    }
+
+    @Override
+    public String toString(Document document) {
```
Не нужно

lyaska commented on src/main/java/ru/documents/factory/OutgoingFactory.java:
```diff
@@ -0,0 +1,30 @@
+        document.setDestination(generateFio());
+        document.setDelivery(generateDelivery());
+        return document;
+
```
пропуски

lyaska commented on src/main/java/ru/documents/factory/TaskFactory.java:
```diff
@@ -0,0 +1,33 @@
+
+
+    @Override
+    public String toString(Document document) {
```
не нужен

lyaska commented on src/main/java/ru/documents/Document.java:
```diff
@@ -0,0 +1,82 @@
+package ru.documents;
+
+import java.util.Comparator;
+/**  класс родитель для видов документов*/
```
ctrl+alt+l

lyaska commented on src/main/java/ru/documents/Document.java:
```diff
@@ -0,0 +1,82 @@
+public class Document {
+    int id;
+    String name;
+    String text;
```
Напиши javadoc для каждого поля в каждом классе

lyaska commented on src/main/java/ru/documents/Document.java:
```diff
@@ -0,0 +1,82 @@
+
+    // реализация интерфейса Comparator
+    class DocumentRegNumComparator implements Comparator<Document> {
+        public int compare(Document a, Document b) {
```
Вложенный класс не нужен

lyaska commented on src/main/java/ru/documents/DocumentExistException.java:
```diff
@@ -0,0 +1,8 @@
+package ru.documents;
+/**  выбрасывает исключение, если документ с генерируемым номером уже существует*/
+public class DocumentExistException extends Exception {
```
ctrl+alt+l

lyaska commented on src/main/java/ru/documents/Incoming.java:
```diff
@@ -0,0 +1,52 @@
+    public void setOutgoing_date(String outgoing_date) {
+        this.outgoing_date = outgoing_date;
+    }
+
```
Лишний пропуск

lyaska commented on src/main/java/ru/documents/Person.java:
```diff
@@ -0,0 +1,50 @@
+
+/**
+ * этот класс понадобится позже
+ */
```
Напиши нормальный джавадок, ты же знаешь что это за класс и зачем он нужен

lyaska commented on src/main/java/MainClass.java:
```diff
@@ -0,0 +1,37 @@
+import java.util.HashMap;
+
+/**
+ * цикл для создания документов, в консоль выводится мапа
```
В консоль выводится не мапа, а список документов. Класс вообще запускает приложение

lyaska commented on src/main/java/MainClass.java:
```diff
@@ -0,0 +1,37 @@
+public class MainClass {
+    public static void main(String[] args) {
+        HashMap<String, ArrayList<Document>> personAndDocument = new HashMap<String, ArrayList<Document>>();
+        RandomValue randomValue = new RandomValue();
```
Не нужно создавать экземпляры класса, сделай методы randomValue static и вызывай по имени класса

lyaska commented on src/main/java/MainClass.java:
```diff
@@ -0,0 +1,37 @@
+            DocumentCreate documentCreate = randomValue.generateRandomFactory();
+            Document doc = documentCreate.generateRandomPropertiesForDocument(documentCreate.create());
+            listDocument[i] = doc;
+            ArrayList<Document> docList = personAndDocument.get(randomValue.listFio[i]);
```
коллекции документов и юзеров пусть будут в отдельном классе, private static, так же вызывай по имени класса

BespoyasovaV replied:
Я создала класс CollectionsOfPersonAndDocuments, 
public class CollectionsOfPersonAndDocument {
    public static Map<String, ArrayList<Document>> personAndDocument = new TreeMap<String, ArrayList<Document>>();
     public static Document[] listDocument = new Document[RandomValue.generateRandomNumber(9)];
}
вынесла туда коллекции, но в main вызвать через класс не могу

lyaska commented on src/main/java/MainClass.java:
```diff
@@ -0,0 +1,37 @@
+        for (int i = 0; i < randomValueForDocList; i++) {
+            DocumentCreate documentCreate = randomValue.generateRandomFactory();
+            Document doc = documentCreate.generateRandomPropertiesForDocument(documentCreate.create());
+            listDocument[i] = doc;
```
добавляй сгенереные документы к персону в отдельном классе. Не бойся создавать новые классы, лучше много маленьких классов чем один большой и для всего. Класс main должен только запускать цепочку создания документов и выводить их в консоль

lyaska commented on src/main/java/MainClass.java:
```diff
@@ -0,0 +1,37 @@
+        }
+        System.out.println(randomValueForDocList);
+        System.out.println(personAndDocument);
+        System.out.println(listDocument.length);
```
что за лишние выводы? Должны быть только Person-ы и их список документов

lyaska commented on src/main/java/ru/generate/RandomValue.java:
```diff
@@ -0,0 +1,135 @@
+/**
+ * класс для генерации рандомных значений
+ */
+public class RandomValue {
```
сделай класс final, от него не нужно наследоваться, а конструктор приватным

lyaska commented on src/main/java/ru/generate/RandomValue.java:
```diff
@@ -0,0 +1,135 @@
+ */
+public class RandomValue {
+    int regNumForTask = 0;
+    int regNumForIncoming = 0;
```
зачем нужны эти поля?

lyaska commented on src/main/java/ru/generate/RandomValue.java:
```diff
@@ -0,0 +1,135 @@
+    }
+
+    public int generateRandomNumber() {
+        int rand_number = (int) (Math.random() * 7);
```
можно подключить библиотеку ApacheCommonsLang и использовать RandomUtils для генерации случайных значений, вызывать метод по классу RandomUtils прямо в сеттере и тогда многие методы из этого класса можно будет удалить

lyaska commented on src/main/java/ru/generate/RandomValue.java:
```diff
@@ -0,0 +1,135 @@
+    }
+
+    public boolean generateSign() {
+        int sign = (int) (Math.random() * 100);
```
Apache Commons Lang

lyaska commented on src/main/java/ru/generate/RandomValue.java:
```diff
@@ -0,0 +1,135 @@
+    }
+
+    public boolean generateController() {
+        int sign = (int) (Math.random() * 100);
```
Сделай все методы статическими

lyaska commented on src/main/java/ru/generate/RandomValue.java:
```diff
@@ -0,0 +1,135 @@
+
+    public String generateNameDoc() {
+        int rand1 = (int) (Math.random() * one);
+        int rand2 = (int) (Math.random() * two);
```
Ты теперь знаешь про дженерики, напиши единый статический метод, который будет доставать любой элемент из любой коллекции.

lyaska commented on src/main/java/ru/generate/RandomValue.java:
```diff
@@ -0,0 +1,135 @@
+
+    public int generateTerm() {
+        int term = (int) (Math.random() * 56);
+
```
пропуски

lyaska commented on src/main/java/ru/generate/RandomValue.java:
```diff
@@ -0,0 +1,135 @@
+    }
+
+    public int generateId() {
+        int id = (int) (Math.random() * 100);
```
Вызывай в фабрике при создании документа UUID.randomUUID и этот метод не нужен будет

lyaska commented on src/main/java/ru/documents/Document.java:
```diff
@@ -0,0 +1,82 @@
+import java.util.Comparator;
+/**  класс родитель для видов документов*/
+public class Document {
+    int id;
```
Можно использовать UUID, почитай что это, дальше будет работа с бд, а сейчас везде такой формат айдишника используется

lyaska commented on src/main/java/ru/generate/RandomValue.java:
```diff
@@ -0,0 +1,135 @@
+    ArrayList<Person> personList = new ArrayList<>();
+    ArrayList<Integer> check_id = new ArrayList<Integer>();
+    String[] arr = new String[]{"Task", "Incoming", "Outgoing"};
+    String[] wordListOne = new String[]{"Доменная", "Схематичная", "Базовая", "Первичная"};
```
имхо, лучше коллекции.

lyaska commented on src/main/java/MainClass.java:
```diff
@@ -0,0 +1,37 @@
+ * цикл для создания документов, в консоль выводится мапа
+ */
+public class MainClass {
+    public static void main(String[] args) {
```
запуши свой pom.xml тоже

lyaska commented on src/main/java/ru/documents/Document.java:
```diff
@@ -0,0 +1,82 @@
+
+import java.util.Comparator;
+/**  класс родитель для видов документов*/
+public class Document {
```
Убери классы документов и юзеров в пакет типа entity, не нужно им просто так валяться

lyaska commented on src/main/java/ru/documents/DocumentExistException.java:
```diff
@@ -0,0 +1,8 @@
+public class DocumentExistException extends Exception {
+
+    public DocumentExistException() {
+        System.out.println("Документ с генерируемым номером уже существует");
```
public DocumentExistsException(String description) {
        super(description);
    }

lyaska commented on src/main/java/ru/generate/RandomValue.java:
```diff
@@ -0,0 +1,135 @@
+        return date1;
+    }
+
+    public DocumentCreate generateRandomFactory() {
```
Пусть этот класс будет только для генерации случайных значений, а этот метод вынеси в другой final утилитный класс

BespoyasovaV committed:
JavaDoc, author, правильно заносятся документы в map, вывод, как в образце.
методы toString() на своих местах, вместо денерации id класс UUID.

lyaska commented on src/main/java/ru/documents/entity/Incoming.java:
```diff
@@ -0,0 +1,72 @@
+    }
+
+    public String toString() {
+        return name_first + " " + "№" + " " + getReg_num() + " " + "от" + " " + getDate_reg() + " " + getName();
```
используй MessageFormat

lyaska commented on src/main/java/ru/documents/entity/Incoming.java:
```diff
@@ -0,0 +1,72 @@
+     * Название вида документа
+     */
+    String name_first = "Входящий";
+    ArrayList<Integer> check_id = new ArrayList<Integer>();
```
Что за коллекция?

BespoyasovaV replied:
больше не нужна, уберу

lyaska commented on src/main/java/ru/documents/factory/DocumentCreate.java:
```diff
@@ -0,0 +1,20 @@
+import ru.documents.DocumentExistException;
+
+/**
+ * От этого класса наследуются фабрики, которые оздают документы разных видов
```
Это интерфейс, классы его реализуют. Не наследуются

BespoyasovaV committed:
метод compareTo() я переделала(правильно?), и класс исключения вместе с самим исключением, теперь оно обрабатывается, а не пробрасывается.
Вопрос про даты, может есть метод, который генерирует рандомные даты? У меня они сейчас вообще String и это явно неправильно.

BespoyasovaV committed:
Исправила поле date и ее генерацию, документы теперь сортируются по регистрационному номеру или(если они совпадают) по дате.
Убрала лишние циклы из main, вынесла коллекции в отдельный класс. Появились методы длы выывода и занесения докумкентов мапу

BespoyasovaV committed:
JavaDoc, author, правильно заносятся документы в map, вывод, как в образце.
методы toString() на своих местах, вместо денерации id класс UUID.

BespoyasovaV committed:
Убрала лишние циклы из main(теперь они в классе CollectionsForAllYourNeeds), собрала все коллекции в один класс, enum тоже в отдельном классе. Метод compareTo() Теперь сортирует документы по регистрационному номеру и по дате  регистрации, вывод,как в примере.

BespoyasovaV committed:
Убрала лишние циклы из main(теперь они в классе CollectionsForAllYourNeeds), собрала все коллекции в один класс, enum тоже в отдельном классе. Метод compareTo() Теперь сортирует документы по регистрационному номеру и по дате  регистрации, вывод,как в примере.
Исправила поля даты и ее генерацию.

BespoyasovaV committed:
Убрала лишние циклы из main(теперь они в классе CollectionsForAllYourNeeds), собрала все коллекции в один класс, enum тоже в отдельном классе. Метод compareTo() Теперь сортирует документы по регистрационному номеру и по дате  регистрации, вывод,как в примере.
Исправила поля даты и ее генерацию.

lyaska commented on src/main/java/ru/documents/entity/Document.java:
```diff
@@ -0,0 +1,100 @@
+ *
+ * @author BespoyasovaV
+ */
+public class Document<T> implements Comparable<Document> {
```
А зачем тут дженерик?

dreambot92 replied:
Вероятнее всего не потребуется создавать экземпляры класса Document. Сделай его abstract

lyaska commented on src/main/java/ru/documents/entity/Document.java:
```diff
@@ -0,0 +1,100 @@
+    }
+
+    public int getReg_num() {
+        return reg_num;
```
camelcase

lyaska commented on src/main/java/ru/documents/entity/Incoming.java:
```diff
@@ -0,0 +1,77 @@
+
+    public String toString() {
+        int month = getDate_reg().getMonth() + 1;
+        int year = getDate_reg().getYear() + 1900;
```
Реализуй метод через MessageFormat. "Входящий №... от .... ..имя.."

lyaska commented on src/main/java/ru/documents/factory/AbstractFactory.java:
```diff
@@ -0,0 +1,32 @@
+            document.setText(generateNameText());
+            document.setReg_num(generateRandomNumber(100));
+        } catch (DocumentExistException e) {
+            System.out.println(e.getMessage());
```
давай вместо этого использовать логгер

lyaska commented on src/main/java/ru/generate/CollectionsForAllYourNeeds.java:
```diff
@@ -0,0 +1,65 @@
+public final class CollectionsForAllYourNeeds {
+    public static TreeMap<String, ArrayList<Document>> personAndDocument = new TreeMap<>();
+    /**
+     * хдесь хранятся все созданные документы без разбора
```
опечатка

lyaska commented on src/main/java/ru/generate/GenerateRandomFactory.java:
```diff
@@ -0,0 +1,25 @@
+import ru.documents.factory.OutgoingFactory;
+import ru.documents.factory.TaskFactory;
+
+public class GenerateRandomFactory {
```
джавадоки

lyaska commented on src/main/java/ru/documents/DocumentExistException.java:
```diff
@@ -0,0 +1,12 @@
+ *
+ * @author BespoyasovaV
+ */
+public class DocumentExistException extends Exception {
```
положив отдельный пакет exceptions

lyaska commented on src/main/java/ru/generate/CollectionsForAllYourNeeds.java:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    /**
+     * отображает отсортироанных по алфавиту авторов и их документы
```
опечатка

lyaska commented on src/main/java/ru/generate/CollectionsForAllYourNeeds.java:
```diff
@@ -0,0 +1,65 @@
+     * отображает отсортироанных по алфавиту авторов и их документы
+     */
+    public static void conclusion() {
+        for (String key : personAndDocument.keySet()) {
```
messageFormat

lyaska commented on src/main/java/ru/persons/Department.java:
```diff
@@ -0,0 +1,45 @@
+ *
+ * @author BespoyasovaV
+ */
+public class Department extends Staff {
```
Это что за классы?

lyaska commented on src/main/java/ru/persons/Staff.java:
```diff
@@ -0,0 +1,5 @@
+package ru.persons;
+
+public class Staff {
```
что за класс?

dreambot92 commented on src/main/java/ru/documents/entity/Document.java:
```diff
@@ -0,0 +1,100 @@
+    /**
+     * Идентификатор документа
+     */
+    UUID id;
```
изменить модификатор доступа на скрытный

dreambot92 commented on src/main/java/ru/documents/entity/Document.java:
```diff
@@ -0,0 +1,100 @@
+    /**
+     * Регистрационный номер документа
+     */
+    int reg_num;
```
по код-стайлу предполагается именновать переменные через camelCase

dreambot92 commented on src/main/java/ru/documents/entity/Document.java:
```diff
@@ -0,0 +1,100 @@
+     */
+    String name;
+    /**
+     * Текст  документа
```
лишний пробел после Текст

dreambot92 commented on src/main/java/ru/documents/entity/Document.java:
```diff
@@ -0,0 +1,100 @@
+     */
+    String author;
+
+    //getters
```
придирочка.. обычно не принято обозначать комментами, где находятся геттеры, сеттеры

dreambot92 commented on src/main/java/ru/documents/entity/Document.java:
```diff
@@ -0,0 +1,100 @@
+
+    @Override
+    public int compareTo(Document o) {
+        if (this.reg_num - o.reg_num < 0) {
```
номер регистрационный уникальный, поэтому в сортировке сначала по номеру, потом по дате мало смысла. В задании имелось ввиду сначала сортировать по дате.. а уже потом по номеру

albertino90 replied:
Используй Сomparator.comparing

albertino90 commented on src/main/java/ru/documents/entity/Document.java:
```diff
@@ -0,0 +1,100 @@
+ *
+ * @author BespoyasovaV
+ */
+public class Document<T> implements Comparable<Document> {
```
Сделай класс Document абстрактным

dreambot92 commented on src/main/java/ru/documents/entity/Document.java:
```diff
@@ -0,0 +1,100 @@
+    public int compareTo(Document o) {
+        if (this.reg_num - o.reg_num < 0) {
+            return -1;
+        }
```
кстати compareTo для целых чисел можно сделать проще
Integer.compare(x, y)

dreambot92 commented on src/main/java/ru/documents/factory/DocumentCreate.java:
```diff
@@ -0,0 +1,19 @@
+
+/**
+ * От этого класса  наследуются фабрики, которые оздают документы разных видов
+ */
```
@author

dreambot92 commented on src/main/java/ru/generate/CollectionsForAllYourNeeds.java:
```diff
@@ -0,0 +1,65 @@
+
+import ru.documents.entity.Document;
+
+import java.util.*;
```
по code style стоит использовать конкретные import-ы.. без звездочек

albertino90 commented on src/main/java/ru/documents/entity/Document.java:
```diff
@@ -0,0 +1,100 @@
+    public void setId(UUID id) {
+        this.id = id;
+    }
+
```
Equals и hashcode

dreambot92 commented on src/main/java/ru/generate/CollectionsForAllYourNeeds.java:
```diff
@@ -0,0 +1,65 @@
+ * @author BespoyasovaV
+ */
+public final class CollectionsForAllYourNeeds {
+    public static TreeMap<String, ArrayList<Document>> personAndDocument = new TreeMap<>();
```
При объявлении переменных лучше использовать более общие интерфейсы, чтобы в случае чего поменять реализацию. Имеется ввиду лучше написать так
Map<String, Collection<Document>> personAndDocument = new TreeMap<>();

dreambot92 replied:
и лучше назвать как-нибудь так personDocuments

albertino90 commented on src/main/java/ru/documents/entity/Task.java:
```diff
@@ -0,0 +1,89 @@
+
+    public String toString() {
+        int month = getDate_reg().getMonth() + 1;
+        int year = getDate_reg().getYear() + 1900;
```
Используй MessageFormat.format

dreambot92 commented on src/main/java/ru/generate/CollectionsForAllYourNeeds.java:
```diff
@@ -0,0 +1,65 @@
+    /**
+     * хдесь хранятся все созданные документы без разбора
+     */
+    public static List<Document> listDocument = new ArrayList<>();
```
побольше javadoc-ов.. не понятно что за коллекции

dreambot92 replied:
здесь лучше documents

dreambot92 commented on src/main/java/ru/generate/CollectionsForAllYourNeeds.java:
```diff
@@ -0,0 +1,65 @@
+    static {
+        listFio = Arrays.asList("Барсуков Сергей Геннадиевич", "Березнянкова Анфиса Кирилловна",
+                "Киреева Надежда Семенова", "Никитина Юлия Игоревна", "Полянская Кристина Викторовна");
+        arr = Arrays.asList("Task", "Incoming", "Outgoing");
```
Типы документов имеет смысл перенести в отдельный Enum

albertino90 replied:
Типы документов лучше вынеси в ENUM

dreambot92 commented on src/main/java/ru/generate/CollectionsForAllYourNeeds.java:
```diff
@@ -0,0 +1,65 @@
+     * метод заносит документы из listDocument в мапу и сортирует по авторам
+     */
+    public static void populate() {
+        Collections.sort(CollectionsForAllYourNeeds.listDocument);
```
если использовать TreeSet, то не нужно будет лишний раз сортировать

dreambot92 commented on src/main/java/ru/generate/CollectionsForAllYourNeeds.java:
```diff
@@ -0,0 +1,65 @@
+ *
+ * @author BespoyasovaV
+ */
+public final class CollectionsForAllYourNeeds {
```
может как-нибудь по другому назовем?
IntershipCollections, DocumentCollections, ProgramCollections

dreambot92 commented on src/main/java/ru/generate/CollectionsForAllYourNeeds.java:
```diff
@@ -0,0 +1,65 @@
+    /**
+     * метод заносит документы из listDocument в мапу и сортирует по авторам
+     */
+    public static void populate() {
```
имхо, стоит перенести логику по работе с этими коллекциями в отдельный сервис

dreambot92 commented on src/main/java/ru/generate/Delivery.java:
```diff
@@ -0,0 +1,12 @@
+ * @author BespoyasovaV
+ */
+public enum Delivery {
+    САМОВЫВОЗ,
```
так можно да,  но давай лучше сделаем через латиницу))
можно использовать enum с параметрами, чтобы хранить локаль
COURIER("Курьер")

dreambot92 commented on src/main/java/ru/generate/CollectionsForAllYourNeeds.java:
```diff
@@ -0,0 +1,65 @@
+    public static ArrayList<Integer> arrException = new ArrayList<>();
+
+    static {
+        listFio = Arrays.asList("Барсуков Сергей Геннадиевич", "Березнянкова Анфиса Кирилловна",
```
это больше нужно для сервиса RandomValue, может имеет смысл хранить именно там

dreambot92 replied:
то есть отделить бы может разными местами хранения коллекции с рандомными и захардкоженными значениями и коллекцию людей и документов

albertino90 commented on src/main/java/ru/generate/GenerateRandomFactory.java:
```diff
@@ -0,0 +1,25 @@
+        int generateRandomNum = (int) (Math.random() * 3);
+        DocumentCreate documentCreate = null;
+        if (CollectionsForAllYourNeeds.arr.get(generateRandomNum).equals("Task")) {
+            documentCreate = new TaskFactory();
```
Сделай через switch-case

dreambot92 commented on src/main/java/ru/persons/Organization.java:
```diff
@@ -0,0 +1,41 @@
+package ru.persons;
+/** Наследник класса Staff,нужен для создания организаций
```
здесь javadoc-и поехали

dreambot92 commented on src/main/java/ru/persons/Person.java:
```diff
@@ -0,0 +1,62 @@
+package ru.persons;
+/**
```
еще один enter

dreambot92 commented on target/maven-status/maven-compiler-plugin/compile/default-compile/createdFiles.lst:
```diff
@@ -0,0 +1,19 @@
+ru\documents\factory\OutgoingFactory.class
```
target, как и файлы идеи присутствовать в ветке не должны

BespoyasovaV replied:
как из убрать?

dreambot92 commented on .idea/compiler.xml:
```diff
@@ -10,4 +10,9 @@
       </profile>
     </annotationProcessing>
   </component>
+  <component name="JavacSettings">
```
.gitignore надо бы добавить и удалить лишние файлы с ветки

BespoyasovaV replied:
и это как?

BespoyasovaV committed:
MessageFormat реализовала, логгер тоже,опечатки убрала, в отделтный пакет исключение вынесла, Document сделала абсрактным,Equals и Hashcode появились,типы документов вынесла в ENUM,создание рандомной фабрикик через switch-case,методы из класса с колеекциями вынесла в отдельный.
А вот с тестами беда.. Зачем вообще тестировать, правильно ли из коллекции берется значение?

BespoyasovaV committed:
Сделала тесты на фабрики, исправили логгер.

BespoyasovaV committed:
08/02/2021

BespoyasovaV committed:
08/02/2021

BespoyasovaV committed:
08/02/2021

BespoyasovaV committed:
08/02/2021
